### PR TITLE
NCNN GPU selection fix

### LINKED ISSUE #932: Wrong enumeration of GPU's in NCNN settings
**Information:**
- Chainner version: 0.12.4
- OS: Windows 11

**Description**

My Dedicated GPU is `0` - NVIDIA GeForce 3060 laptop
While my integrated GPU is `1`.

However, chaiNNer associates their names wrong. If I select option 0 it will run on NVIDIA, while if I choose 1 it runs on integrated AMD.

### PR DESCRIPTION
**Pull request by joeyballentine**
Requires #947's NCNN update to work properly

fixes #932

### CODE REVIEW

**Review by iamwavecut — APPROVED** [review]
Great! :)

**Comment by Diwizard80** [comment]
Pls unsubscribe me. I tried every way too, still get emails

Warm Regards,
Reuben Piscopo

IPhotographyJunkie / AI Artist / Sky Watcher / Videography / Universe Wanderer / Web3 & Metaverse

on Instagram, Linkedin, Vimeo, Twitter, Rarible & Opensea

**Comment by joeyballentine** [comment]
@Diwizard80 modify your GitHub settings and remove emails from your notifications, or click on the eye icon next to the star and fork buttons to stop watching the repo

**Comment by Diwizard80** [comment]
Did it all

Warm Regards,
Reuben Piscopo

IPhotographyJunkie / AI Artist / Sky Watcher / Videography / Universe Wanderer / Web3 & Metaverse

on Instagram, Linkedin, Vimeo, Twitter, Rarible & Opensea

**Comment by joeyballentine** [comment]
@Diwizard80 there isn't anything I can do to "unsubscribe" you myself. Contact GitHub support or something. This is entirely a problem with your own settings

File: src/renderer/components/SettingsModal.tsx [hunk 1]
```diff
@@ -322,15 +324,15 @@ const PythonSettings = memo(() => {
...
     const [ncnnGPU, setNcnnGPU] = useNcnnGPU;
-    const [fullGpuList, setFullGpuList] = useState<string[]>([]);
+    const [ncnnGpuList, setNcnnGpuList] = useState<string[]>([]);
     useAsyncEffect(
         {
             supplier: async () => {
-                const fullGpuInfo = await ipcRenderer.invoke('get-gpu-info');
-                const gpuNames = fullGpuInfo.controllers.map((g) => g.model);
+                const ncnnGpuInfo = await backend.listNcnnGpus();
+                const gpuNames = Object.values(ncnnGpuInfo);
```
Comment (RunDevelopment) [inline src/renderer/components/SettingsModal.tsx:332]: Why did you make the backend return a `Record<number, string>` when you assume it's indexed like a `string[]` here?

However, I'm not sure which of the two is the problem. Is the backend more complex than it needs to be, or is the frontend making an assumption it shouldn't make?

Comment (joeyballentine) [reply src/renderer/components/SettingsModal.tsx:332]: I could have returned a list from the backend, but I wanted to ensure order. I can change this to whatever you think is the proper way

Comment (RunDevelopment) [reply src/renderer/components/SettingsModal.tsx:332]: > but I wanted to ensure order

Aren't lists ordered? Or asked in another way: What order did you want to ensure?

Comment (joeyballentine) [reply src/renderer/components/SettingsModal.tsx:332]: I just wanted to make sure no f*ckery happens between the JSON conversion and parsing. If that's guaranteed to be correct then I'll just use a list. I'm trying to preserve gpu index order

Comment (joeyballentine) [reply src/renderer/components/SettingsModal.tsx:332]: 😂 array it is, then

**Review by RunDevelopment — APPROVED** [review]

**Pull request merged**